File: backend/app/schemas/metricas.py
"""
Schemas para gráficas del módulo Métricas
"""
from pydantic import BaseModel, Field


class DistribucionItem(BaseModel):
    nombre: str
    valor: int


class PuntoFinanzas(BaseModel):
    etiqueta: str
    mes: str
    ventas: float = 0.0
    ventas_leche: float = 0.0
    ventas_animales: float = 0.0
    ventas_otros: float = 0.0
    gastos: float = 0.0
    compras: float = 0.0
    balance: float = 0.0


class PuntoProduccion(BaseModel):
    etiqueta: str
    mes: str
    litros: float = 0.0


class PuntoReproductivo(BaseModel):
    etiqueta: str
    mes: str
    servicios: int = 0
    partos: int = 0


class PuntoConciliacionLeche(BaseModel):
    etiqueta: str
    mes: str
    litros_ordeñados: float = 0.0
    litros_vendidos: float = 0.0
    diferencia: float = 0.0
    ingreso_leche: float = 0.0


class ResumenMargenRubro(BaseModel):
    ingresos_leche: float = 0.0
    gastos_leche: float = 0.0
    margen_leche: float = 0.0
    ingresos_ceba: float = 0.0
    gastos_ceba: float = 0.0
    margen_ceba: float = 0.0
    gastos_general: float = 0.0


class MetricasGraficas(BaseModel):
    meses: int = Field(description="Ventana temporal usada en las series")
    inventario_categorias: list[DistribucionItem] = []
    inventario_estados: list[DistribucionItem] = []
    finanzas: list[PuntoFinanzas] = []
    produccion: list[PuntoProduccion] = []
    reproductivo: list[PuntoReproductivo] = []
    conciliacion_leche: list[PuntoConciliacionLeche] = []
    margen_rubros: ResumenMargenRubro = Field(default_factory=ResumenMargenRubro)

File: frontend/src/api/metricas.ts
import apiClient from './client';

export interface DistribucionItem {
  nombre: string;
  valor: number;
}

export interface PuntoFinanzas {
  etiqueta: string;
  mes: string;
  ventas: number;
  ventas_leche: number;
  ventas_animales: number;
  ventas_otros: number;
  gastos: number;
  compras: number;
  balance: number;
}

export interface PuntoProduccion {
  etiqueta: string;
  mes: string;
  litros: number;
}

export interface PuntoReproductivo {
  etiqueta: string;
  mes: string;
  servicios: number;
  partos: number;
}

export interface PuntoConciliacionLeche {
  etiqueta: string;
  mes: string;
  litros_ordeñados: number;
  litros_vendidos: number;
  diferencia: number;
  ingreso_leche: number;
}

export interface ResumenMargenRubro {
  ingresos_leche: number;
  gastos_leche: number;
  margen_leche: number;
  ingresos_ceba: number;
  gastos_ceba: number;
  margen_ceba: number;
  gastos_general: number;
}

export interface MetricasGraficas {
  meses: number;
  inventario_categorias: DistribucionItem[];
  inventario_estados: DistribucionItem[];
  finanzas: PuntoFinanzas[];
  produccion: PuntoProduccion[];
  reproductivo: PuntoReproductivo[];
  conciliacion_leche: PuntoConciliacionLeche[];
  margen_rubros: ResumenMargenRubro;
}

export const metricasService = {
  async getMetricas(meses: number = 6): Promise<MetricasGraficas> {
    const response = await apiClient.get('/dashboard/metricas', { params: { meses } });
    return response.data;
  },
};

export default metricasService;
